docs/getting_started_jax.ipynb

```diff
@@ -0,0 +1,669 @@
+            "module @jit_f attributes {mhlo.num_partitions = 8 : i32, mhlo.num_replicas = 1 : i32} {\n",
+            "  func.func public @main(%arg0: tensor<32x64xf32>) -> (tensor<32x64xf32> {jax.result_info = \"\"}) {\n",
+            "    %0 = stablehlo.custom_call @Sharding(%arg0) {backend_config = \"unspecified_dims=[1]\", mhlo.sharding = \"{devices=[4,1,2]<=[8] last_tile_dim_replicate}\"} : (tensor<32x64xf32>) -> tensor<32x64xf32>\n",
+            "    return %0 : tensor<32x64xf32>\n",
+            "  }\n",
+            "}\n",
+            "\n"
+          ]
+        }
+      ],
+      "source": [
+        "def run_with_sharding_constraint():\n",
+        "  x = jax.ShapeDtypeStruct((32, 64), jnp.float32)\n",
+        "\n",
+        "  def f(x):\n",
+        "    return jax.lax.with_sharding_constraint(x, NamedSharding(mesh, PartitionSpec('data', PartitionSpec.UNCONSTRAINED)))\n",
+        "\n",
+        "  print(jax.jit(f).lower(x).as_text())\n",
+        "\n",
+        "run_with_sharding_constraint()"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "d5pLLzvsbf6O"
+      },
+      "source": [
+        "But under Shardy it's an explicit op:"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 8,
+      "metadata": {
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        },
+        "id": "haybWp_0bxI8",
+        "outputId": "e25d9fdc-23da-4f35-8cd9-e61f0d7938ba"
+      },
+      "outputs": [
+        {
+          "name": "stdout",
+          "output_type": "stream",
+          "text": [
+            "module @jit_f attributes {mhlo.num_partitions = 8 : i32, mhlo.num_replicas = 1 : i32} {\n",
+            "  sdy.mesh @mesh = <[\"data\"=4, \"model\"=2]>\n",
+            "  func.func public @main(%arg0: tensor<32x64xf32>) -> (tensor<32x64xf32> {jax.result_info = \"\"}) {\n",
+            "    %0 = sdy.sharding_constraint %arg0 <@mesh, [{\"data\"}, {?}]> : tensor<32x64xf32>\n",
+            "    return %0 : tensor<32x64xf32>\n",
+            "  }\n",
+            "}\n",
+            "\n"
+          ]
+        }
+      ],
+      "source": [
+        "jax.config.update(\"jax_use_shardy_partitioner\", True)\n",
+        "run_with_sharding_constraint()\n",
+        "jax.config.update(\"jax_use_shardy_partitioner\", False)"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "kXxwCyk6KbT-"
+      },
+      "source": [
+        "Note that `UNCONSTRAINED` under GSPMD has the custom call have an op attribute `backend_config = \"unspecified_dims=[1]\"`. But under Shardy, it makes dim 1 be `{?}`. In Shardy, dimension shardings without a `?` are closed, meaning that dimension can't be further sharded, but when it has a trailing `?`, it can be further sharded. Refer to [Sharding representation](https://github.com/openxla/shardy/blob/add-docs/docs/sharding_representation.md) for more info on the sharding representation."
```

Comment:
Double checking is the link `openxla/shardy/blob/add-docs/docs/sharding_representation.md` correct? Should it be `openxla/shardy/docs/sharding_representation.md` or be `openxla/shardy/tree/main/docs/sharding_representation.md`?

Reply:
Yes, sorry about that!